hollow/src/main/java/com/netflix/hollow/api/producer/validation/RecordCountChangeValidator.java

```diff
@@ -0,0 +1,125 @@
+        log.info(validationResult.toString());
+        return validationResult;
+    }
+
+    private ValidationResult validateChanges(HollowTypeReadState typeState) {
+        if(typeState.getPreviousOrdinals().isEmpty()) {
+            return ValidationResult.from(this).passed("Ignore the check if previous records are empty.");
+        }
+        int addRecordNumber = accessor.getAddedRecords().size();
+        int removeRecordNumber = accessor.getRemovedRecords().size();
+        int updatedRecordNumber = accessor.getUpdatedRecords().size();
+        int previousRecordNumber = typeState.getPreviousOrdinals().cardinality();
+
+        float addedPercent = (float) addRecordNumber / previousRecordNumber;
+        float removedPercent = (float) removeRecordNumber / previousRecordNumber;
+        float updatedPercent = (float) updatedRecordNumber / previousRecordNumber;
+
+
+        float addedPercentageThreshold = threshold.addedPercentageThreshold;
+        float removedPercentageThreshold = threshold.removedPercentageThreshold;
+        float updatedPercentageThreshold = threshold.updatedPercentageThreshold;
+
+        ValidationResult.ValidationResultBuilder builder = ValidationResult.from(this);
+        builder.detail("addedRecordNumber", addRecordNumber);
+        builder.detail("removedRecordNumber", removeRecordNumber);
+        builder.detail("updatedRecordNumber", updatedRecordNumber);
+        builder.detail("previousRecordNumber", previousRecordNumber);
+        builder.detail("addedPercentageThreshold", addedPercentageThreshold);
+        builder.detail("removedPercentageThreshold", removedPercentageThreshold);
+        builder.detail("updatedPercentageThreshold", updatedPercentageThreshold);
+
+        boolean pass =
+                (addedPercentageThreshold < 0 || addedPercent < addedPercentageThreshold) &&
+                        (removedPercentageThreshold < 0 || removedPercent < removedPercentageThreshold) &&
+                        (updatedPercentageThreshold < 0 || updatedPercent < updatedPercentageThreshold);
+        if (pass) {
+            return builder.passed();
+        }
+        return builder.failed("record count change is more than threshold");
+    }
+
+    public static class Threshold {
+        private final float removedPercentageThreshold;
+        private final float addedPercentageThreshold;
+        private final float updatedPercentageThreshold;
+
+        Threshold(float removedPercentageThreshold,
+                  float addedPercentageThreshold,
+                  float updatedPercentageThreshold) {
+            this.removedPercentageThreshold = removedPercentageThreshold;
+            this.addedPercentageThreshold = addedPercentageThreshold;
+            this.updatedPercentageThreshold = updatedPercentageThreshold;
+        }
+
+        public static ThresholdBuilder builder() {
+            return new ThresholdBuilder();
+        }
+
+        public static class ThresholdBuilder {
```

Comment:
if a use case using the builder pattern only supplies the deletion threshold, i think we treat it as- they dont care about added/updated percent. Those should be pass through.

Reply:
This seems already handed since we're initializing the thresholds to -1 👍

Reply:
maybe this usage can also be documented in javadoc